x/autopilot/module_ibc.go

```diff
@@ -203,7 +203,12 @@ func (im IBCModule) OnRecvPacket(
 				im.keeper.Logger(ctx).Error(fmt.Sprintf("Error liquid staking packet from autopilot for %s: %s", newData.Sender, err.Error()))
 				return channeltypes.NewErrorAcknowledgement(err)
 			}
-			// case types.RedeemStake: TODO: add redeem stake logic
+		case types.RedeemStake:
+			// Try to redeem stake - return an ack error if it fails, otherwise return the ack generated from the earlier packet propogation
+			if err := im.keeper.TryRedeemStake(ctx, packet, newData, routingInfo); err != nil {
```

Comment:
just checking - if this fails, do the tokens end up the receiver address on Stride? If so, it might be worth adding an integration test checking this (just to triple-check we avoid the worst-case scenario of this failing and the tokens getting burned)

Reply:
+1 to Vishal's comment, afaiu this would only require adding a single check to the existing integration test

Reply:
@sampocs it looks like this is the only comment that hasn't been addressed (let me know if that's not the case, and thank you for picking up the rest 🙏 )

should I update the integration test for this?

Reply:
> if this fails, do the tokens end up the receiver address on Stride?

If this fails, we get an ack error and the whole inbound transfer fails, sending the tokens back to the host zone.

I can update the integration test since you're hopping offline rn!